docs/source/tutorials/using_text.rst

```diff
@@ -0,0 +1,461 @@
+:class:`~.Text` class. For example, the ``color`` keyword changes the
+color of the TeX mobject.
+
+.. manim:: LaTeXAttributes
+    :save_last_frame:
+
+    class LaTeXAttributes(Scene):
+        def construct(self):
+            tex = Tex(r'Hello \LaTeX', color=BLUE).scale(3)
+            self.add(tex)
+
+Extra LaTeX Packages
+====================
+
+Some commands require special packages to be loaded into the TeX template. 
+For example, to use the ``mathscr`` script, we need to add the ``mathrsfs``
+package. Since this package isn't loaded into Manim's tex template by default,
+we have to add it manually.
+
+.. manim:: AddPackageLatex
+    :save_last_frame:
+
+    class AddPackageLatex(Scene):
+        def construct(self):
+            myTemplate = TexTemplate()
+            myTemplate.add_to_preamble(r"\usepackage{mathrsfs}")
+            tex = Tex(r'$\mathscr{H} \rightarrow \mathbb{H}$}', tex_template=myTemplate).scale(3)
+            self.add(tex)
+
+Substrings and parts
+====================
+
+The TeX mobject can accept multiple strings as arguments. Afterwards you can
+refer to the individual parts either by their index (like ``tex[1]``), or by
+selecting parts of the tex code. In this example, we set the color
+of the ``\bigstar`` using :func:`~.set_color_by_tex`:
+
+.. manim:: LaTeXSubstrings
+    :save_last_frame:
+
+    class LaTeXSubstrings(Scene):
+        def construct(self):
+            tex = Tex('Hello', r'$\bigstar$', r'\LaTeX').scale(3)
+            tex.set_color_by_tex('igsta', RED)
+            self.add(tex)
+
+Note that :func:`~.set_color_by_tex` colors the entire substring containing
+the Tex, not just the specific symbol or Tex expression. Consider the following example:
+
+.. manim:: IncorrectLaTeXSubstringColoring
+    :save_last_frame:
+
+    class IncorrectLaTeXSubstringColoring(Scene):
+        def construct(self):
+            equation = MathTex(
+                r"e^x = x^0 + x^1 + \frac{1}{2} x^2 + \frac{1}{6} x^3 + \cdots + \frac{1}{n!} x^n + \cdots"
+            )
+            equation.set_color_by_tex("x", YELLOW)
+            self.add(equation)
+
+As you can see, this colors the entire equation yellow, contrary to what 
+may be expected. To color only ``x`` yellow, we have to do the following:
+
+.. manim:: CorrectLaTeXSubstringColoring
+    :save_last_frame:
+
+    class CorrectLaTeXSubstringColoring(Scene):
+        def construct(self):
+            equation = MathTex(
+                r"e^x = x^0 + x^1 + \frac{1}{2} x^2 + \frac{1}{6} x^3 + \cdots + \frac{1}{n!} x^n + \cdots",
+                substrings_to_isolate="x"
+            )
+            equation.set_color_by_tex("x", YELLOW)
+            self.add(equation)
+
+By setting ``substring_to_isolate`` to ``x``, we split up the
+:class:`~.MathTex` into substrings automatically and isolate the ``x`` components 
+into individual substrings. Only then can :meth:`~.set_color_by_tex` be used 
+to achieve the desired result.
+
```

Comment:
I suggest to add the following paragraph on the custom `{{...}}` syntax that Manim supports.

```suggestion

Note that Manim also supports a custom syntax that allows splitting
a TeX string into substrings easily: simply enclose parts of your formula
that you want to isolate with double braces. In the string
``MathTex(r"{{ a^2 }} + {{ b^2 }} = {{ c^2 }}")``, the rendered mobject
will consist of the substrings ``a^2``, ``+``, ``b^2``, ``=``, and ``c^2``.
This makes transformations between similar text fragments easy
to write using :class:`~.TransformMatchingTex`.

```